### PR TITLE
Use micro-frontend method instead of iframe

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Render the cards using the react script from dashboard instead of iframes.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: wp-content/plugins/hurumap/package.json [hunk 1]
```diff
@@ -33,10 +33,11 @@
     "start": "concurrently \"yarn start:data-page\"  \"yarn start:blocks\"",
     "start:blocks": "wp-scripts start ./src/FlourishBlock/index.js ./src/HURUmapBlock/index.js ./src/FeaturedData/index.js ./src/CardBlock/index.js --output-path=build/blocks",
     "start:data-page": "wp-scripts start ./src/index.js --output-path=build/data-page",
-    "build": "yarn build:data-page && yarn build:blocks && yarn build:embed",
+    "build": "yarn build:data-page && yarn build:blocks && yarn build:micro-frontend",
     "build:data-page": "wp-scripts build ./src/index.js --output-path=build/data-page",
     "build:blocks": "wp-scripts build ./src/FlourishBlock/index.js ./src/HURUmapBlock/index.js ./src/FeaturedData/index.js ./src/CardBlock/index.js --output-path=build/blocks",
-    "build:embed": "wp-scripts build ./src/embed/index.js -o ../../themes/hurumap/assets/build/embed.js"
+    "build:micro-frontend": "wp-scripts build ./src/micro-frontend/hurumap-ui.js -o ../../themes/hurumap/micro-frontend/build/hurumap-ui.js --config webpack.micro-frontend.config.js",
+    "build:e": "wp-scripts build ./src/embed/e.js -o ../../themes/hurumap/assets/build/e.js"
```
Comment (KhadijaMahanga) [inline wp-content/plugins/hurumap/package.json]: `build: e`  .. clean up

**Review by KhadijaMahanga — COMMENTED** [review]
- [x] can we rename `hurumap-ui.js`

File: wp-content/plugins/hurumap/package.json [hunk 2]
```diff
@@ -33,10 +33,11 @@
     "start": "concurrently \"yarn start:data-page\"  \"yarn start:blocks\"",
     "start:blocks": "wp-scripts start ./src/FlourishBlock/index.js ./src/HURUmapBlock/index.js ./src/FeaturedData/index.js ./src/CardBlock/index.js --output-path=build/blocks",
     "start:data-page": "wp-scripts start ./src/index.js --output-path=build/data-page",
-    "build": "yarn build:data-page && yarn build:blocks && yarn build:embed",
+    "build": "yarn build:data-page && yarn build:blocks && yarn build:micro-frontend",
     "build:data-page": "wp-scripts build ./src/index.js --output-path=build/data-page",
     "build:blocks": "wp-scripts build ./src/FlourishBlock/index.js ./src/HURUmapBlock/index.js ./src/FeaturedData/index.js ./src/CardBlock/index.js --output-path=build/blocks",
-    "build:embed": "wp-scripts build ./src/embed/index.js -o ../../themes/hurumap/assets/build/embed.js"
+    "build:micro-frontend": "wp-scripts build ./src/micro-frontend/hurumap-ui.js -o ../../themes/hurumap/micro-frontend/build/hurumap-ui.js --config webpack.micro-frontend.config.js",
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/package.json]: Lets call it `hurumap-ui-microfrontend` or something then..

File: wp-content/plugins/hurumap/package.json [hunk 3]
```diff
@@ -83,4 +84,4 @@
       "git add"
     ]
   }
-}
+}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/package.json]: `\n`

File: wp-content/plugins/hurumap/src/CardBlock/Save.js [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
+    <div
       id={`hurumap-card-${attributes.id}`}
-      style={{ marginLeft: 10, marginBottom: 10 }}
-      title={`${attributes.postType}-${attributes.postId}`}
-      src={`${config.WP_BACKEND_URL}/card/${attributes.postType}/${attributes.postId}?width=${attributes.cardWidth}&id=${attributes.id}`}
+      style={{
+        marginLeft: 10,
+        marginBottom: 10,
+        width: attributes.cardWidth
+      }}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/src/CardBlock/Save.js:12]: `class` won't work? Harder to override styles defined using `style`

File: wp-content/plugins/hurumap/src/CardBlock/Save.js [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
+      data-postType={attributes.postType}
+      data-postId={attributes.postId}
+      data-width={attributes.cardWidth}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/src/CardBlock/Save.js:15]: `data-post-type`, `data-post-id`: https://stackoverflow.com/a/25033330

Browser auto-lowercase could work but...

File: wp-content/plugins/hurumap/src/micro-frontend/hurumap-ui.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  );
+}
+
+Card.propTypes = {
+  postId: propTypes.string.isRequired,
+  postType: propTypes.string.isRequired,
+  parentEl: propTypes.element.isRequired
+};
+
+window.onload = () => {
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/src/micro-frontend/hurumap-ui.js]: Cool, but I prefer `window.addEventListener('load'` just to be safe

**Review by kilemensi — APPROVED** [review]
:rocket: @karimkawambwa

File: wp-content/plugins/hurumap/src/CardBlock/Save.js (hunk 4 above)
Comment (karimkawambwa) [reply wp-content/plugins/hurumap/src/CardBlock/Save.js:12]: I'll resolve this in another PR

**Pull request merged**